Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -394,6 +389,11 @@ void AppendPipeline::appsinkNewSample(const Track& track, GRefPtr<GstSample>&& s
 
     auto mediaSample = MediaSampleGStreamer::create(WTFMove(sample), track.presentationSize, track.trackId);
 
+    if (track.editListOffset) {
+        MediaTime presentationTime = mediaSample->presentationTime() - MediaTime(GST_TIME_AS_USECONDS(track.editListOffset), G_USEC_PER_SEC);
```

Comment:
There might be a precision loss here? Perhaps use nanoseconds?

Reply:
I tried to keep the precision by using `MediaTime(track.editListOffset, GST_SECOND)` and the whole test times out during a seek. That's because most of the conversions from GstClockTime to MediaTime in the code are done with [fromGstClockTime()](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h#L95), which internally does the same lossy conversion I did.
I'm going to resubmit a new version of the patch calling `fromGstClockTime()`. At least, if the default precision policy changes in the future, my change goes along with it.